discord/channel.py

```diff
@@ -1351,6 +1356,198 @@ async def create_thread(
+            Specifies the slowmode rate limit for user in this channel, in seconds.
+            A value of `0` disables slowmode. The maximum value possible is `21600`.
+        reason: Optional[:class:`str`]
+            The reason for editing this channel. Shows up on the audit log.
+        overwrites: Dict[Union[:class:`Role`, :class:`Member`, :class:`~discord.abc.Snowflake`], :class:`PermissionOverwrite`]
+            The overwrites to apply to channel permissions. Useful for creating secret channels.
+        default_auto_archive_duration: :class:`int`
+            The new default auto archive duration in minutes for threads created in this channel.
+            Must be one of ``60``, ``1440``, ``4320``, or ``10080``.
+        default_thread_slowmode_delay: :class:`int`
+            The new default slowmode delay in seconds for threads created in this channel.
+
+        default_sort_order: Optional[:class:`SortOrder`]
+            The default sort order type to use to order posts in this channel.
+
+        default_reaction_emoji: Optional[:class:`discord.GuildEmoji` | :class:`int` | :class:`str`]
+            The default reaction emoji.
+            Can be a unicode emoji or a custom emoji in the forms:
+            :class:`GuildEmoji`, snowflake ID, string representation (e.g., '<a:emoji_name:emoji_id>').
+
+        available_tags: List[:class:`ForumTag`]
+            The set of tags that can be used in this channel. Must be less than `20`.
+
+        require_tag: :class:`bool`
+            Whether a tag should be required to be specified when creating a thread in this channel.
+
+        hide_media_download_options: :class:`bool`
+            Whether media download options should be hidden in this media channel.
+
+        Returns
+        -------
+        Optional[:class:`.MediaChannel`]
+            The newly edited media channel. If the edit was only positional
+            then ``None`` is returned instead.
+
+        Raises
+        ------
+        InvalidArgument
+            If position is less than 0 or greater than the number of channels, or if
+            the permission overwrite information is not in proper form.
+        Forbidden
+            You do not have permissions to edit the channel.
+        HTTPException
+            Editing the channel failed.
+        """
+        with contextlib.suppress(KeyError):
+            require_tag = options.pop("require_tag")
+            options["flags"] = options.get("flags") or ChannelFlags._from_value(
+                self.flags.value
+            )
+            options["flags"].require_tag = require_tag
+
+        with contextlib.suppress(KeyError):
+            hide_media_download_options = options.pop("hide_media_download_options")
+            options["flags"] = options.get("flags") or ChannelFlags._from_value(
+                self.flags.value
+            )
+            options["flags"].hide_media_download_options = hide_media_download_options
```

Comment:
prettier, probably
we also don't need to check if "flags" is set in options because, well, it shouldn't be (it's not documented)
```suggestion
        if "require_tag" in options or "hide_media_download_options" in options:
            flags = ChannelFlags._from_value(self.flags.value)
            flags.require_tag = options.pop("require_tag", flags.require_tag)
            flags.hide_media_download_options = options.pop("hide_media_download_options", flags.hide_media_download_options)
            options["flags"] = flags
            
```